controllers/nodehealthcheck_controller.go

```diff
@@ -326,17 +326,25 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	for _, node := range healthyNodes {
 		log.Info("handling healthy node", "node", node.GetName())
 		remediationCRs, err := resourceManager.ListRemediationCRs(nhc, func(cr unstructured.Unstructured) bool {
-			return cr.GetName() == node.GetName()
+			return cr.GetName() == node.GetName() && resources.IsOwner(&cr, nhc)
 		})
 		if err != nil {
 			log.Error(err, "failed to get remediation CRs for healthy node", "node", node.Name)
 			return result, err
 		}
-		for _, remediationCR := range remediationCRs {
-			if err := r.deleteRemediationCR(&remediationCR, nhc, resourceManager, log); err != nil {
-				log.Error(err, "failed to delete remediation CR", "name", remediationCR.GetName())
-				return result, err
-			}
+
+		if len(remediationCRs) == 0 {
+			// when all CRs are gone, the node is considered healthy
+			resources.UpdateStatusNodeHealthy(node.GetName(), nhc, r.Recorder)
+			continue
+		}
+
+		// don't count nodes with remediation CRs as healthy
+		fixedHealthy := *nhc.Status.HealthyNodes - 1
+		nhc.Status.HealthyNodes = &fixedHealthy
```

Comment:
I'm not sure updating the HealthyNodes is correct for this use case.
I guess it depends on what we consider a healthy node.
In case (as implied by the code) we consider a healthy node: a node that passed successfully `checkNodeHealth`  we might want to use a different (and new) indicator for nodes that we consider healthy but has an existing remediation

Reply:
what about adding a `conditionsHealthy` timestamp in status.unhealthyNodes[] for visibility?

Reply:
>I understood that's the issue: counting nodes as healthy which still have a CR 🤷🏼‍♂️

IMO it is fine counting nodes that passed healthcheck as healthy, even though I do acknowledge why it's confusing.
I think that in general there are two valid paths here.
- we don't consider nodes with remediation healthy even if they passed the health check
In that case we take that path:
1.  some of the code terminology needs to be changed as well (for example we shouldn't use var name 
`healthyNodes` for nodes we consider unhealthy)
2. We need to make some distinction between such nodes and nodes that didn't pass the health check (IIUC your followup comment was a suggestion on a way to achieve that)

- we consider nodes healthy if the passed the health check
In that case we take that path: 
We need to make some distinction between such nodes with remediation and those without (in order to mitigate the confusion).

Personally I'm leaning more into the second path, but if you feel strongly you prefer the first one (or maybe another option I didn't consider) that's fine too.

Reply:
I went with the 1st path, with the proposed change in status.unhealthyNodes and better naming of vars and methods.

- nodes are not really healthy, while they most likely still are unschedulable, or miss other cleanup by their remediator
- not counting those nodes as healthy is prudent from the perspective of allowing subsequent remediations